tests/errortests.cpp

```diff
@@ -145,14 +145,61 @@ namespace parser_load {
   }
 }
 
+namespace adversarial {
+  bool number_overrun_at_root() {
+    TEST_START();
+    constexpr const char *json = "122222222222222222222222222222222222222,";
+    constexpr size_t len = 1; // strlen("1");
+    static_assert(strlen(json)-len > SIMDJSON_PADDING); // Make sure it's big enough to overrun
+
+    dom::parser parser;
+    uint64_t foo;
+    ASSERT_SUCCESS( parser.parse(json, len).get(foo) ); // Parse just the first digit
+    ASSERT_EQUAL( foo, 1 );
+    TEST_SUCCEED();
+  }
+  bool number_overrun_in_array() {
+    TEST_START();
+    constexpr const char *json = "[122222222222222222222222222222222222222]";
+    constexpr size_t len = 2; // strlen("[1");
+    static_assert(strlen(json)-len > SIMDJSON_PADDING); // Make sure it's big enough to overrun
+
+    dom::parser parser;
+    uint64_t foo;
+    ASSERT_ERROR( parser.parse(json, len).get(foo), TAPE_ERROR ); // Parse just the first digit
+    TEST_SUCCEED();
+  }
+  bool number_overrun_in_object() {
+    TEST_START();
+    constexpr const char *json = "{\"key\":122222222222222222222222222222222222222}";
+    constexpr size_t len = 8; // strlen("{\"key\":1");
+    static_assert(strlen(json)-len > SIMDJSON_PADDING); // Make sure it's big enough to overrun
```

Comment:
This cannot be a static_assert because Visual Studio won't let you consider strlen(json) to be a compile-time constant.